apps/mobile/src/screens/Onboarding/OnboardingUsernameScreen.tsx

```diff
@@ -0,0 +1,211 @@
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import clsx from 'clsx';
+import { useColorScheme } from 'nativewind';
+import { useCallback, useEffect, useState } from 'react';
+import { KeyboardAvoidingView, Platform, TextInput, View } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+import { BackButton } from '~/components/BackButton';
+import { Button } from '~/components/Button';
+import { Typography } from '~/components/Typography';
+import { LoginStackNavigatorParamList, LoginStackNavigatorProp } from '~/navigation/types';
+import { useReportError } from '~/shared/contexts/ErrorReportingContext';
+import useCreateUser from '~/shared/hooks/useCreateUser';
+import useDebounce from '~/shared/hooks/useDebounce';
+import { useIsUsernameAvailableFetcher } from '~/shared/hooks/useUserInfoFormIsUsernameAvailableQuery';
+import colors from '~/shared/theme/colors';
+import {
+  alphanumericUnderscores,
+  maxLength,
+  minLength,
+  noConsecutivePeriodsOrUnderscores,
+  required,
+  validate,
+} from '~/shared/utils/validators';
+
+import useSyncTokens from '../NftSelectorScreen/useSyncTokens';
+
+export function OnboardingUsernameScreen() {
+  const navigation = useNavigation<LoginStackNavigatorProp>();
+  const { colorScheme } = useColorScheme();
+  const createUser = useCreateUser();
+  const isUsernameAvailableFetcher = useIsUsernameAvailableFetcher();
+  const reportError = useReportError();
+  const { isSyncing, syncTokens } = useSyncTokens();
+
+  const route = useRoute<RouteProp<LoginStackNavigatorParamList, 'OnboardingUsername'>>();
+
+  const [username, setUsername] = useState('');
+  const [bio] = useState('');
+
+  // This cannot be derived from a "null" `usernameError`
+  // since the value starts off as empty when the form is empty
+  const [isUsernameValid, setIsUsernameValid] = useState(false);
+  const [usernameError, setUsernameError] = useState('');
+
+  const debouncedUsername = useDebounce(username, 500);
+
+  const authMechanism = route.params.authMechanism;
+
+  const { top, bottom } = useSafeAreaInsets();
+
+  const handleBack = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  const handleUsernameChange = useCallback((text: string) => {
+    setUsernameError('');
+    setUsername(text);
+  }, []);
+
+  const handleNext = useCallback(async () => {
+    setUsernameError('');
+
+    try {
+      const response = await createUser(authMechanism, username, bio);
+
+      if (response.createUser?.__typename === 'CreateUserPayload') {
+        if (!isSyncing) {
+          syncTokens('Ethereum');
```

Comment:
question:
- if the UI auto-syncs when user signs up, will it be clear on the tokens screen that it's _currently syncing_ and the user is unable to click the refresh button again? (sorry i could probably check this myself lol)
- note: with incremental syncing, we should make a note that you don't need to click Refresh as the tokens stream in. no action needed, will coordinate with benny!

Reply:
@Robinnnnn 

1. There is no indicator the token is currently syncing and the user able to click refresh again! Should we disabled it the refresh button
2. Got it! Added comment to removed it when incremental syncing implemented